features/nanostack/mbed-mesh-api/mbed-mesh-api/mesh_interface_types.h

```diff
@@ -64,6 +64,93 @@ typedef enum {
+    uint32_t ip_no_route;           /*<! IP no route count. */
+    /* Fragments */
+    uint32_t frag_rx_errors;        /*<! Fragmentation RX error count. */
+    uint32_t frag_tx_errors;        /*<! Fragmentation TX error count. */
+    /*RPL stats*/
+    uint32_t rpl_route_routecost_better_change; /*<! RPL parent change count. */
+    uint32_t ip_routeloop_detect;               /*<! RPL route loop detection count. */
+    uint32_t rpl_memory_overflow;   /*<! RPL memory overflow count. */
+    uint32_t rpl_parent_tx_fail;    /*<! RPL transmit errors to DODAG parents. */
+    uint32_t rpl_unknown_instance;  /*<! RPL unknown instance ID count. */
+    uint32_t rpl_local_repair;      /*<! RPL local repair count. */
+    uint32_t rpl_global_repair;     /*<! RPL global repair count. */
+    uint32_t rpl_malformed_message; /*<! RPL malformed message count. */
+    uint32_t rpl_time_no_next_hop;  /*<! RPL seconds without a next hop. */
+    uint32_t rpl_total_memory;      /*<! RPL current memory usage total. */
+    /* Buffers */
+    uint32_t buf_alloc;             /*<! Buffer allocation count. */
+    uint32_t buf_headroom_realloc;  /*<! Buffer headroom realloc count. */
+    uint32_t buf_headroom_shuffle;  /*<! Buffer headroom shuffle count. */
+    uint32_t buf_headroom_fail;     /*<! Buffer headroom failure count. */
+    /* ETX */
+    uint16_t etx_1st_parent;        /*<! Primary parent ETX. */
+    uint16_t etx_2nd_parent;        /*<! Secondary parent ETX. */
+    /* MAC */
+    uint16_t adapt_layer_tx_queue_size; /*<! Adaptation layer direct TX queue size. */
+    uint16_t adapt_layer_tx_queue_peak; /*<! Adaptation layer direct TX queue size peak. */
+    /* Wi-SUN */
+    uint32_t asynch_tx_count;  /*<! Asynch TX counter */
+    uint32_t asynch_rx_count;  /*<! Asynch RX counter */
+} mesh_nw_statistics_t;
+
+/**
+ * Mesh physical layer statistics
+ */
+typedef struct {
+    uint32_t crc_fails;    /*<! CRC failures */
+    uint32_t tx_timeouts;  /*<! transmission timeouts */
+    uint32_t rx_timeouts;  /*<! reception timeouts */
+} mesh_phy_statistics_t;
+
+/**
+ * Mesh MAC statistics
+ */
+typedef struct {
+    uint16_t mac_tx_queue_size;       /*<! MAC TX queue current size. */
+    uint16_t mac_tx_queue_peak;       /*<! MAC TX queue peak size. */
+    uint32_t mac_rx_count;            /*<! MAC RX packet count. */
+    uint32_t mac_tx_count;            /*<! MAC TX packet count. */
+    uint32_t mac_bc_rx_count;         /*<! MAC broadcast RX packet count. */
+    uint32_t mac_bc_tx_count;         /*<! MAC broadcast TX packet count. */
+    uint32_t mac_beacon_rx_count;     /*<! MAC Beacon RX packet count. */
+    uint32_t mac_beacon_tx_count;     /*<! MAC Beacon TX packet count. */
+    uint32_t mac_rx_drop_count;       /*<! MAC RX packet drop count. */
+    uint32_t mac_tx_bytes;            /*<! MAC TX bytes count. */
+    uint32_t mac_rx_bytes;            /*<! MAC RX bytes count. */
```

Comment:
We have multiple levels of RX counts and byte counts I think one is enough
Mac_tx/rx_count
Mac_bx_tx/rx_count and 
Mac_tx/rx/bytes 
seems the best on that area

Reply:
corrected